includes/class-wc-ajax.php

```diff
@@ -431,8 +457,48 @@ public static function remove_from_cart() {
 	 */
 	public static function checkout() {
 		wc_maybe_define_constant( 'WOOCOMMERCE_CHECKOUT', true );
-		WC()->checkout()->process_checkout();
-		wp_die( 0 );
+
+		if ( $_COOKIE['woocommerce_cart_dirty'] !== '0' ) {
+			wc_add_notice( __( 'Cart has been updated. Please check the order again.', 'woocommerce' ) );
+
+			// Get order review fragment
```

Comment:
This is a lot of duplication from `update_order_review` - can they not share the same protected method to grab these fragments and return JSON?